staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -108,6 +113,9 @@ func RegisterStorageCleanup(fn func()) {
 }
 
 func CleanupStorage() {
+	if atomic.AddInt64(&trackStorageCleanupAtomic, -1) != 0 {
```

Comment:
Why not just modify this once you acquire the cleanupLock?

Reply:
Also please add a comment explaining why this is required. I was surprised to see this change in a test PR.

Reply:
Can we just modify `TearDownFn` to optionally not tear down the storage?

Reply:
Since we need to keep a count of the initialization and cleanups, I figured taking the lock is redundant. It's possible it might deadlock within the lock as well.

Reply:
It's just really intrusive to introduce this to the regular storage support the integration test framework. I'd prefer we add something to `cmd/kube-apiserver/app/testing/` instead.

Reply:
Oh wait, I didn't see the `HACK ALERT` above :(

I guess keeping it here s fine. Please leave a comment explaining why this counter's required.